src/components/ItemList/index.jsx

```diff
@@ -0,0 +1,65 @@
+import React, { useState, useEffect } from 'react';
+import { updateItemDate } from '../Utils/firestore';
+import { differenceInHours } from 'date-fns';
+
+function ItemList({ itemName, docId, lastPurchasedDate }) {
+  const [isChecked, setIsChecked] = useState(false);
+  const [isDisabled, setIsDisabled] = useState(false);
+  const [token] = useState(() => window.localStorage.getItem('tcl18-token'));
+  const toggle = () => setIsChecked(!isChecked);
+
+  const editProduct = (id) => {
+    updateItemDate(token, id)
+      .then(() => {
+        console.log('product updated');
+      })
+      .catch((error) => {
+        console.log(error);
+      });
+  };
+
+  useEffect(() => {
+    if (isChecked) {
+      updateItemDate(token, docId).then(() =>
+        console.log('The date was updated'),
+      );
+    }
+  }, [isChecked, docId, token]);
+
+  useEffect(() => {
+    if (lastPurchasedDate && isDisabled === false) {
+      const hoursAtTheMoment = differenceInHours(
+        new Date(),
+        lastPurchasedDate.toDate(),
+      );
+      setIsDisabled(hoursAtTheMoment <= 12);
+      setIsChecked(hoursAtTheMoment <= 12);
+    }
+  }, [isDisabled, lastPurchasedDate]);
+
+  const handleCheckbox = (event) => {
+    toggle();
```

Comment:
Can you explain what is the goal with `toggle()` function?

Reply:
that function manages the state of the checkboxes;

Reply:
I would remove the toggle function and just call 
setIsChecked inside this function. It is cleaner and easier to read.

Reply:
The fix was implemented.

Reply:
Thanks for your explanation! 😄